spec/rubocop/cop/rspec/aggregate_examples_spec.rb

```diff
@@ -0,0 +1,790 @@
+      let(:cop_config) do
+        { 'MatchersWithSideEffects' => [] }
+      end
+
+      offensive_source = <<-RUBY
+        describe 'aggregations' do
+          it { is_expected.to validate_absence_of(:comment) }
+          it { is_expected.to validate_presence_of(:description) }
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Aggregate with the example above.
+        end
+      RUBY
+
+      good_source = <<-RUBY
+        describe 'aggregations' do
+          specify do
+            is_expected.to validate_absence_of(:comment)
+            is_expected.to validate_presence_of(:description)
+          end
+        end
+      RUBY
+
+      it_behaves_like 'detects and autocorrects', offensive_source, good_source
+    end
+
+    context 'with `is_expected`' do
+      offensive_source = <<-RUBY
+        describe 'aggregations' do
+          it { is_expected.to validate_absence_of(:comment) }
+          it { is_expected.to validate_presence_of(:description) }
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Aggregate with the example above. IMPORTANT! Pay attention to the expectation order, some of the matchers have side effects.
+        end
+      RUBY
+
+      it_behaves_like 'detects, but does not autocorrect', offensive_source
+    end
+
+    context 'with `expect(something)`' do
+      offensive_source = <<-RUBY
+        describe 'aggregations' do
+          it { expect(something).to validate_absence_of(:comment) }
+          it { expect(something).to validate_presence_of(:description) }
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Aggregate with the example above. IMPORTANT! Pay attention to the expectation order, some of the matchers have side effects.
+        end
+      RUBY
+
+      it_behaves_like 'detects, but does not autocorrect', offensive_source
+    end
+
+    context 'with `not_to`' do
+      offensive_source = <<-RUBY
+        describe 'aggregations' do
+          it { is_expected.not_to validate_absence_of(:comment) }
+          it { is_expected.to_not validate_presence_of(:description) }
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Aggregate with the example above. IMPORTANT! Pay attention to the expectation order, some of the matchers have side effects.
+        end
+      RUBY
+
+      it_behaves_like 'detects, but does not autocorrect', offensive_source
+    end
+  end
+
+  context 'with validation actions with side effects' do
+    offensive_source = <<-RUBY
+      describe do
+        describe 'without side effects' do
+          it { is_expected.to be_good }
+          it { is_expected.to be_cheap }
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Aggregate with the example above.
+          context 'with side effects' do
+            it { is_expected.to validate_presence_of(:comment) }
+          end
+        end
+      end
+    RUBY
+
+    it 'detects an offense in offensive_source code' do
+      expect_offense(offensive_source)
```

Comment:
I think the autocorrection result would be interesting in this example. I’ve seen above how only the first offensive line would be marked with `^^^`, so I wonder if the `validate_presence_of` line is actually considered offensive or not.